pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -118,6 +122,42 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 	return defaultNwCIDRs, nil
 }
 
+// getNodeDefaultLabels returns true if the node has labels for subnet and Pod range
+func getNodeDefaultLabels(node *v1.Node) (bool, string, string) {
+	defaultSubnet, foundSubnet := node.Labels[utilnode.NodePoolSubnetLabelPrefix]
+	defaultPodRange, foundRange := node.Labels[utilnode.NodePoolPodRangeLabelPrefix]
+	if !foundSubnet || defaultSubnet == "" || !foundRange || defaultPodRange == "" {
+		return false, "", ""
+	}
+	return true, defaultSubnet, defaultPodRange
+}
+
+// getDefaultNwCIDRs returns the Pod CIDRs for default Network.
+// Different subnet can have the same secondary range name, here uses the subnet and range name
+// to find the matching CIDR(s)
+func (ca *cloudCIDRAllocator) getDefaultNwCIDRs(node *v1.Node, interfaces []*compute.NetworkInterface, defaultSubnet, defaultPodRange string) (defaultNwCIDRs []string) {
+out:
+	for _, inf := range interfaces {
+		// extra the subnetwork name from the URL
+		parts := strings.Split(inf.Subnetwork, "/subnetworks/")
+		if parts[1] != defaultSubnet {
+			continue
+		}
+		for _, ipRange := range inf.AliasIpRanges {
+			if ipRange.SubnetworkRangeName != defaultPodRange {
+				continue
+			}
+			defaultNwCIDRs = append(defaultNwCIDRs, ipRange.IpCidrRange)
+			ipv6Addr := ca.cloud.GetIPV6Address(inf)
+			if ipv6Addr != nil {
+				defaultNwCIDRs = append(defaultNwCIDRs, ipv6Addr.String())
+			}
+			break out
```

Comment:
Why are we going back to the outer for loop?

Reply:
The `break out` will break both the inner loop and outer loop.